en/debug/gdb_debugging.md

```diff
@@ -1,127 +1,65 @@
-# Embedded Debugging
+# Debugging with GDB
 
-## Handy console commands
+The [GNU DeBugger (GDB)](https://sourceware.org/gdb/download/onlinedocs/gdb/index.html) comes installed with the compiler toolchain in the form of the `arm-none-eabi-gdb` binary.
+The debugger reads the debug symbols inside an ELF file to understand the static and dynamic memory layout of the PX4 firmware.
+To access the PX4 autopilot microcontroller, it needs to connect to a [Remote Target](https://sourceware.org/gdb/download/onlinedocs/gdb/Connecting.html), which is provided by a [SWD debug probe](swd_debug).
 
-Below are a couple of commands which can be used in the [NuttShell](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=139629410) to get insights of the system.
+The flow of information looks like this:
 
-This NSH command provides the remaining free memory:
-
-```bash
-free
+```
+Developer <=> GDB <=> GDB Server <=> Debug Probe <=> SWD <=> PX4 Autopilot.
 ```
 
-And the top command shows the stack usage per application:
+## Quickstart
 
-```
-top
-```
+To start a debugging session you typically:
 
-Stack usage is calculated with stack coloring and thus is not the current usage, but the maximum since the start of the task.
+1. Need a specialized [SWD debug probe](swd_debug#debug-probes).
```

Comment:
The link works but the anchor doesn't - you've pushed all the probes into their own docs.  
1. We probably need a list of probes in that page, each linking to the actual probe - right?
2. The page does have debug adaptors. Should we pull them out as peers of the probes too?

Reply:
```suggestion
1. Need a specialized [SWD debug probe](../debug/swd_debug.md#debug-probes).
```